projects/plugins/boost/app/assets/src/js/features/critical-css/lib/critical-css-errors.ts

```diff
@@ -21,15 +20,46 @@ export type ErrorSet = {
 	};
 };
 
-export const criticalCssIssues = derived( criticalCssState, $status => {
-	return $status.providers.filter( provider => provider.errors?.length > 0 );
-} );
+/**
+ * Given a Critical CSS State, returns whether or not this represents a fatal error.
+ *
+ * @param {CriticalCssState} cssState - The CSS State object.
+ */
+export function isFatalError( cssState: CriticalCssState ): boolean {
+	if ( cssState.status === 'error' ) {
+		return true;
+	}
+
+	if ( cssState.status === 'not_generated' ) {
+		return false;
+	}
+
+	return ! cssState.providers.some( provider =>
+		[ 'success', 'pending' ].includes( provider.status )
+	);
+}
 
 /**
- * Derived datastore: Returns the most important Set of errors among the recommendations.
- * Used for displaying the most important error as a showstopper if no URLS succeeded.
+ * Given a CSS State object, returns all the providers that have errors.
+ *
+ * @param cssState - The CSS State object.
  */
-export const primaryErrorSet: Readable< ErrorSet > = derived( criticalCssIssues, $issues => {
+export function getCriticalCssIssues( cssState: CriticalCssState ): Provider[] {
+	return cssState.providers.filter( provider => provider.errors?.length || 0 > 0 );
```

Comment:
`|| 0 > 0`

`O_O`

....took me a while. Is prettier preventing from wrapping that in parenthesis?

Reply:
Looks like prettier didn't complain 4e58135190